### DISCUSSION TITLE
"Predefined Asset Metadata" vs "Predefined Properties"

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#17177] [Q&A / Support]
What is the difference between "Predefined Asset Metadata" and "Predefined Properties" (when used for Assets)? I could not find any documentation comparing these two.

**Comment by markus-moser** [comment]
I agree that to some extent they are quite similar. Here are the main differences:

- Properties can be inherited to children assets within a folder whereas asset metadata are always related to a single asset.
- Properties can also be added to folders.
- Asset metadata can be localized whereas properties cannot.
- Asset metadata can be shown and batch edited in the asset list/grid view whereas properties cannot.
- Properties are also available for data objects and documents.

The idea behind asset metadata is that they are used (as the name indicates) to save metadata of assets (like for example the title, alt text, copyright, ...) whereas properties are more used for "settings". 

Properties are in my opinion mostly relevant for documents, where this can be for example very helpful to define a snippet (for example the footer of a website) at root level and then it gets inherited to all children and therefore the footer applies to all pages. If you then need a different footer on a single document you can just break inheritance on this level.

But also for assets there are use cases, but this depends highly on your concrete requirements. I used it for example one time to mark all assets which should be exported to another system with a flag. There the inheritance feature is very helpful because you only need to define it at the most top relevant folder level.

**Reply by cancan101** [reply]
Is there any way to filter / search for a given asset based on either of metadata or a property? All I see in the UI is filtering by "tags".
It looks like you can add a column just for the metadata and filter based on it, though that filtering seems to work just for text fields and not any of the relation fields.

**Reply by markus-moser** [reply]
No for relation fields this currently does not work. The only way is the grid column configuration, there it works for some of the field types (e.g. date filter).
